tslearn/docs/shapelets.rst

```diff
@@ -0,0 +1,121 @@
+.. _shapelets:
+
+Shapelets
+=========
+
+Shapelets are defined in [1]_ as "subsequences that are in some sense
+maximally representative of a class".
+Informally, if we assume a binary classification setting, a shapelet is
+discriminant if it is **present** in most series of one class and absent from
+series of the other class.
+To assess the level of presence, one uses shapelet matches:
+
+.. math::
+
+    d(\mathbf{x}, \mathbf{s}) =
+        \min_t \| \mathbf{x}_{t\rightarrow t+L} - \mathbf{s} \|_2
+
+where :math:`L` is the length (number of timestamps) of shapelet
+:math:`\mathbf{s}` and :math:`\mathbf{x}_{t\rightarrow t+L}` is the subsequence
+extracted from time series :math:`\mathbf{x}` that starts at time index
+:math:`t` and stops at :math:`t+L`.
+If the above-defined distance is small enough, then
+shapelet :math:`\textbf{s}` is supposed to be present in time series
+:math:`\mathbf{x}`.
+
+.. figure:: _static/img/shap_distance.png
```

Comment:
Shouldn't we generate our own image here? It would feel more homogeneous to have only code-generated images in our docs, to highlight the fact that Python+tslearn is a good match for time series data analysis, no? We could still have some statement like "inspired from [1]".

Reply:
The same will then apply for my other schematic figure I guess?

Reply:
Yep, maybe, though this other one might require a bit more work. Maybe we could turn that into a gallery example by the way. I kind of like the viz there (taken from the page accompanying the paper): 

![](http://fs.ismll.de/publicspace/LearningShapelets/introShapelet.jpg)

WDYT?